lib/ui/compositing.dart

```diff
@@ -17,6 +17,20 @@ class Scene extends NativeFieldWrapperClass2 {
   /// To create a Scene object, use a [SceneBuilder].
   Scene._();
 
+
+  /// Creates a raster image representation of the current state of the scene.
+  /// This is a slow operation that is performed on a background thread.
+  Future<Image> toImage(int width, int height) {
+    if (width <= 0 || height <= 0) {
+      return null;
+    }
+    return _futurize(
+      (_Callback<Image> callback) => _toImage(width, height, callback)
+    );
+  }
+
+  void _toImage(int width, int height, _Callback<Image> callback) native 'Scene_toImage';
```

Comment:
This should return String, where `nullptr` means success, and a String means an error (where the error message is the return value).

Reply:
I don't think that interface is usable here because I would need to figure out if an error occurred before I returned from the native `Scene_toImage`. Errors could happen on the IO thread too. Or, on the hop back onto the UI thread before the callback is invoked. So, errors are indicated by the presence of a null image in the callback instead.

Reply:
You're getting analyzer errors with it being `null` though, so at least for now, you need to change it to return `String` and just have it always return `nullptr`.

Reply:
I modified the API to return a string. But this will only catch errors on the UI thread. IO thread errors will result in a null Image being returned in the callback.

Reply:
`_futurize()` will detect the callback returning null and will turn that into an error future (with a generic message)

Reply:
Yeah, I realized that the analyzer was getting mad at the usage. I have returned error messages instead. The image can still be null if there are IO thread errors.